lib/eventContext.ts

```diff
@@ -0,0 +1,114 @@
+ * event handler to any of the objects.
+ * @interface EventContext
+ */
+export interface EventContext {
+  /**
+   * @property {Connection} connection The amqp connection.
+   */
+  connection: Connection;
+  /**
+   * @property {Container} container The amqp container
+   */
+  container: Container;
+  /**
+   * @property {Session} [session] The amqp session link that was created on the amqp connection.
+   */
+  session?: Session;
+  /**
+   * @property {Delivery} [delivery] The amqp delivery that is received after sending a message.
+   */
+  delivery?: Delivery;
+  /**
+   * @property {AmqpMessage} [message] The amqp message that is received in the message event
+   * handler when rhea emits a message event on a receiver.
+   */
+  message?: Message;
+  /**
+   * @property {Receiver} [receiver] The amqp receiver link that was created on the amqp connection.
+   */
+  receiver?: Receiver;
+  /**
+   * @property {Sender} [sender] The amqp sender link that was created on the amqp connection.
+   */
+  sender?: Sender;
+  /**
+   * @property {Error | ConnectionError} [error] An optional error object.
+   * - On `connection_error` event this property will be present. It will have the same information as
+   * `connection.error` but the type will be `ConnectionError`.
+   * - An error with SASL will be available through this property, but not through `connection.error`
+   * (as the amqp connection was never established).
+   * - On `disconnected` event the context will have an error property that will be of type
+   * `Error` (or some subclass) as emitted by the underlying socket.
+   * - The `session_error`, `sender_error` and `receiver_error` events will not have this (`error`)
+   * property on the EventContext.
+   */
+  error?: Error | ConnectionError;
+  /**
+   * @property {boolean} [reconnecting] The value is true if the library is attempting to automatically
+   * reconnect and false if it has reached the reconnect limit. If reconnect has not been enabled
+   * or if the connection is a tcp server, then the reconnecting property is undefined. This property
+   * is used in conjunction with "disconnected" event.
+   */
+  reconnecting?: boolean;
+}
+
+export module EventContext {
+  /**
+   * Translates rhea's EventContext into rhea-promise EventContext
+   * @param rheaContext The received context from rhea's event emitter
+   * @param emitter Teh rhea-promise equivalent object that is supposed emit the same event.
+   */
+  export function translate(rheaContext: RheaEventContext, emitter: Link | Session | Connection): EventContext {
+    // initialize the result
+    const result: EventContext = {} as any;
+
+    // clone/copy
+    for (const prop of Object.keys(rheaContext)) {
+      (result as any)[prop] = (rheaContext as any)[prop];
+    }
+
+    const connection: Connection = emitter instanceof Connection
+      ? emitter
+      : (emitter as Link).connection; // both session and link have the connection property.
```

Comment:
nit: might be worth casting as `Link | Session` so that typescript checks the assumption